### ISSUE TITLE
[Feature request] official docker image

### ISSUE THREAD

**Issue by xiaket** [aws-cloudformation/cfn-python-lint#1025]
This will put some extra burden on the developers, especially during the release process, but this will help out the following scenarios:

1. People of no python knowledge can simply try the tool without much effort.
2. Some CI/CD pipelines where the installation of either python or a python package would mean extra work.

Please let me know what you think, and please feel free to turn this down. I'm happy to create a PR if you guys want to go down this path. :)

**Comment by kddejong** [comment]
I have some internal tasks to get this going but wanted to let you know I have started this work.

**Comment by mrpackethead** [comment]
This is a bit of a challenge,  as in some environments using docker images becomes a tricky challenge for security reasons, as we have to pick them all apart to assess if its safe.

**Comment by xiaket** [comment]
IMHO this is a good alternative to the current pip installation method, you can always choose to install through pip if you have other requirements to fulfill. :)

**Comment by gliptak** [comment]
Initial Docker file was merged https://github.com/aws-cloudformation/cfn-python-lint/pull/1361

**Comment by gliptak** [comment]
How would upload to https://hub.docker.com/ happen?

PS There is an unofficial version at https://hub.docker.com/r/aztek/cfn-lint

**Comment by atkinsonm** [comment]
Please publish to DockerHub :pray:

**Comment by jacksgt** [comment]
I agree, an official image on DockerHub (with tags for each release) would be great.
Since the Dockerfile is already in place, just setting up an automatic DockerHub build for the repository will not be too difficult.

Currently, there are many unofficial images out there, but for all of them we cannot know how they are build and how long they will be updated:
https://hub.docker.com/search?q=cfn-python-lint&type=image
https://hub.docker.com/search?q=cfn-lint&type=image

Thus, an official image would be greatly appreciated.
As mentioned in the original issue, it would simplify the setup for many people and also in CI/CD pipelines.

**Comment by scarytom** [comment]
I note that there is an official docker image for the AWS CLI now (see https://github.com/aws/aws-cli/issues/3553#issuecomment-606952933), does this set a precedent that might see an official cfn-lint image?

**Comment by zBart** [comment]
Also added my👍, having to manually create and update a docker image for our build pipelines is quite a hassle.

**Comment by pbudzon** [comment]
+10000 on this. This is a linting tool, using it in CI/CD seems like the main use-case and it's recommended by AWS themselves! Lack of an official docker image is shocking.

**Comment by pbudzon** [comment]
As a public service we have created a public GitHub repository which uses GitHub Actions to **check every day** if a new release of cfn-lint is available and builds an image for it, stored in a public ECR repository:

GitHub repo: https://github.com/MysteriousCode/cfn-lint-docker
ECR repo: https://gallery.ecr.aws/s4a2b7m4/cfn-lint (we're waiting for approval for a proper repository alias from AWS)

If you're not sure you can try this repo, check out the workflow file that builds the image: https://github.com/MysteriousCode/cfn-lint-docker/blob/main/.github/workflows/ecr.yaml

Hope this helps people.
